susemanager-utils/susemanager-sls/salt/packages/profileupdate.sls

```diff
@@ -2,7 +2,7 @@ packages:
   module.run:
     - name: pkg.info_installed
     - kwargs: {
-          attr: 'arch,epoch,version,release,install_date_time_t',
+          attr: 'status,arch,epoch,version,release,install_date_time_t',
```

Comment:
Hmm... I don't really understand this change. The `pkg.info_installed` method should **always only return installed packages** (regardless the `attr` argument).

The `attr` argument only affects to the format of the output, but the list of packages should always be the same (only installed packages).

In case `pkg.info_installed` is reporting uninstalled packages on Ubuntu/Debian, I would expect a fix on the Salt module but not here.

Am I missing something here?

Reply:
You are right. `info_installed` **should** return only installed packages. But since `attr` is just passed to `lowpkg.info`, we are not getting `status` at all. And then this piece of code really shines: `pkg_nfo.get('status', 'ii')`. So if there is not `status` we just assume that it's installed. That's why this is just the quick fix and [this](https://github.com/openSUSE/salt/pull/241/files) will be the proper fix.